src/PKSim.Core/Services/IDiseaseStateImplementation.cs

```diff
@@ -1,56 +1,182 @@
       /// </summary>
       void Validate(OriginData originData);
 
       /// <summary>
-      /// Returns <c>true</c> if the parameters are compatible with th underlying disease state otherwise <c>false</c>
-      /// If the <paramref name="originData"/> is not valid, the return value will contain the reason in error
+      ///    Returns <c>true</c> if the parameters are compatible with th underlying disease state otherwise <c>false</c>
+      ///    If the <paramref name="originData" /> is not valid, the return value will contain the reason in error
       /// </summary>
       (bool isValid, string error) IsValid(OriginData originData);
-      
+
       /// <summary>
-      /// Apply any change required to the disease factor parameter associated with the molecule
+      ///    Apply any change required to the disease factor parameter associated with the molecule
       /// </summary>
-      void ApplyTo(IndividualMolecule individualMolecule);
+      void ApplyTo(Individual individual, IndividualMolecule individualMolecule);
+   }
+
+   public abstract class AbstractDiseaseStateImplementation : IDiseaseStateImplementation
+   {
+      private readonly IIndividualFactory _individualFactory;
+      private readonly IContainerTask _containerTask;
+      private readonly IParameterSetUpdater _parameterSetUpdater;
+      private readonly IFormulaFactory _formulaFactory;
+      private readonly IValueOriginRepository _valueOriginRepository;
+      private readonly string _name;
+
+      protected AbstractDiseaseStateImplementation(
+         IValueOriginRepository valueOriginRepository,
+         IFormulaFactory formulaFactory,
+         IIndividualFactory individualFactory,
+         IContainerTask containerTask,
+         IParameterSetUpdater parameterSetUpdater, string name)
+      {
+         _individualFactory = individualFactory;
+         _containerTask = containerTask;
+         _parameterSetUpdater = parameterSetUpdater;
+         _formulaFactory = formulaFactory;
+         _valueOriginRepository = valueOriginRepository;
+         _name = name;
+      }
+
+      public bool IsSatisfiedBy(DiseaseState diseaseState) => diseaseState.IsNamed(_name);
+
+      public abstract bool ApplyTo(Individual individual);
 
+      public abstract bool ApplyForPopulationTo(Individual individual);
+
+      public Individual CreateBaseIndividualForPopulation(Individual originalIndividual)
+      {
+         //we need to create a new individual WITHOUT CKD and set all percentiles as in the original individuals. Other parameters, value wil be taken as is
+         var originData = originalIndividual.OriginData.Clone();
+
+         //remove the disease state to create a healthy Individual
+         originData.DiseaseState = null;
+         var healthyIndividual = _individualFactory.CreateAndOptimizeFor(originData, originalIndividual.Seed);
+
+         var allCKDParameters = ParameterChangedByDiseaseStateAsList(healthyIndividual);
```

Comment:
rename `allCKDParameters ` to something more generic (`allDeseasedParameters`, whatever)